Subject: [PATCH] Fix tag version for keystone-client

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 82ca730..939a8a6 100644
--- a/setup.py
+++ b/setup.py
@@ -14,7 +14,7 @@ tests_require = setup.parse_requirements(['tools/test-requires'])
 
 setuptools.setup(
     name="python-keystoneclient",
-    version=setup.get_post_version('keystoneclient'),
+    version=setup.get_post_version('keystoneclient').splitlines()[0].strip(),
     description="Client library for OpenStack Identity API (Keystone)",
     long_description=read('README.rst'),
     url='https://github.com/openstack/python-keystoneclient',
-- 
1.7.12

