diff --git a/tests/v1_1/fakes.py b/tests/v1_1/fakes.py
index 91f33e8..95c976f 100644
--- a/tests/v1_1/fakes.py
+++ b/tests/v1_1/fakes.py
@@ -38,6 +38,17 @@ class FakeHTTPClient(base_client.HTTPClient):
         self.password = 'password'
         self.auth_url = 'auth_url'
         self.callstack = []
+        self.projectid = 'projectid'
+        self.user = 'user'
+        self.region_name = 'region_name'
+        self.endpoint_type = 'endpoint_type'
+        self.service_type = 'service_type'
+        self.service_name = 'service_name'
+        self.volume_service_name = 'volume_service_name'
+        self.timings = 'timings'
+        self.bypass_url = 'bypass_url'
+        self.os_cache = 'os_cache'
+        self.http_log_debug = 'http_log_debug'
 
     def _cs_request(self, url, method, **kwargs):
         # Check that certain things are called correctly
