--- keystone-2012.1/keystone/__init__.py.newdeps	2012-02-29 11:16:06.000000000 +0100
+++ keystone-2012.1/keystone/__init__.py	2012-03-01 22:42:08.546503618 +0100
@@ -0,0 +1,29 @@
+import sys
+import pkg_resources
+
+# If there is a conflicting non egg module,
+# i.e. an older standard system module installed,
+# then replace it with this requirement
+def replace_dist(requirement):
+    try:
+        return pkg_resources.require(requirement)
+    except pkg_resources.VersionConflict:
+        e = sys.exc_info()[1]
+        dist=e.args[0]
+        req=e.args[1]
+        if dist.key == req.key and not dist.location.endswith('.egg'):
+            del pkg_resources.working_set.by_key[dist.key]
+            # We assume there is no need to adjust sys.path
+            # and the associated pkg_resources.working_set.entries
+            return pkg_resources.require(requirement)
+
+replace_dist("WebOb >= 1.0")
+replace_dist("SQLAlchemy >= 0.6.3")
+replace_dist("Routes >= 1.12.3")
+
+replace_dist("PasteDeploy >= 1.5.0")
+# This hack is needed because replace_dist() results in
+# the standard paste module path being at the start of __path__.
+# TODO: See can we get pkg_resources to do the right thing directly
+import paste
+paste.__path__.insert(0, paste.__path__.pop(-1))
--- keystone-2012.1/bin/keystone-all.orig   2012-11-09 22:44:31.000000000 +0000
+++ keystone-2012.1/bin/keystone-all    2012-11-09 23:49:04.560249926 +0000
@@ -20,13 +20,13 @@
 
 eventlet.patcher.monkey_patch(all=False, socket=True, time=True, thread=True)
 
-from paste import deploy
-
 from keystone import config
 from keystone.common import wsgi
 from keystone.common import utils
 from keystone.openstack.common import importutils
 
+from paste import deploy
+
 
 CONF = config.CONF
 
