--- a/keystoneclient/openstack/common/setup.py	2013-01-02 11:00:18.207200000 -0800
+++ b/keystoneclient/openstack/common/setup.py	2013-01-02 11:02:09.078872000 -0800
@@ -156,7 +156,8 @@
 
 
 def _get_git_current_tag():
-    return _run_shell_command("git tag --contains HEAD")
+    # return _run_shell_command("git tag --contains HEAD")
+    return _run_shell_command("git describe --always --tag")
 
 
 def _get_git_tag_info():
