Subject: [PATCH] Patch to grab only the last version

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index c3d851c..ad69bf6 100644
--- a/setup.py
+++ b/setup.py
@@ -23,7 +23,7 @@ from quantumclient.openstack.common import setup
 
 Name = 'python-quantumclient'
 Url = "https://launchpad.net/quantum"
-Version = setup.get_post_version('quantumclient')
+Version = setup.get_post_version('quantumclient').splitlines()[-1].strip()
 License = 'Apache License 2.0'
 Author = 'OpenStack Quantum Project'
 AuthorEmail = 'openstack-dev@lists.launchpad.net'
-- 
1.7.12

